From b7cb397fb6eafb64c851b1523869be0f19b73a0a Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Wed, 11 Feb 2015 10:38:34 -0600
Subject: [PATCH 0382/1594] MLK-10247-2: arm: imx: pinctrl imx7d iomuxc LPSR
 pad support

commit 0c8373b6d04592561b9ec4470c648bf8da824111 from
git://git.freescale.com/imx/linux-2.6-imx.git

* Add iomuxc-lpsr device tree node at 302c0000 reg address space
* Add pwm1 pinctrl support pad MX7D_PAD_GPIO1_IO01__PWM1_OUT

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
Signed-off-by: Robin Gong <b38343@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts |   24 ++++++++++++++++++++++++
 arch/arm/boot/dts/imx7d.dtsi           |   10 +++++-----
 2 files changed, 29 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index 6d12164..e2d6cc7 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -152,6 +152,30 @@
 	/* pinctrl-0 = <&pinctrl_uart3dte_1>; */
 };
 
+&iomuxc_lpsr {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_2>;
+
+	hog {
+		pinctrl_hog_2: hoggrp-2 {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO07__GPIO1_IO7	0x80000000
+			>;
+		};
+	};
+};
+
+&iomuxc_lpsr {
+	pwm1 {
+		pinctrl_pwm1: pwm1grp {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO01__PWM1_OUT	0x110b0
+			>;
+		};
+	};
+};
+
+
 &usdhc1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1_1>;
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index a9a6761..d610d0c 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -223,11 +223,6 @@
 				#interrupt-cells = <2>;
 			};
 
-			gpr_lpsr: iomuxc-gpr-lpsr@30270000 {
-				compatible = "fsl,imx7d-iomuxc-lpsr-gpr";
-				reg = <0x30270000 0x10000>;
-			};
-
 			wdog1: wdog@30280000 {
 				compatible = "fsl,imx7d-wdt", "fsl,imx21-wdt";
 				reg = <0x30280000 0x10000>;
@@ -260,6 +255,11 @@
 				status = "disabled";
 			};
 
+			iomuxc_lpsr: iomuxc-lpsr@302c0000 {
+				compatible = "fsl,imx7d-iomuxc-lpsr";
+				reg = <0x302c0000 0x10000>;
+			};
+
 			gpt1: gpt@302d0000 {
 				compatible = "fsl,imx7d-gpt", "fsl,imx31-gpt";
 				reg = <0x302d0000 0x10000>;
-- 
1.7.5.4

